File: app/src/main/java/com/chuongvd/app/signal/data/source/remote/service/BaseErrorResponse.java
package com.chuongvd.app.signal.data.source.remote.service;

import android.text.TextUtils;
import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;
import java.util.List;

public class BaseErrorResponse {

    @SerializedName("code")
    private String code;

    @SerializedName("messageKey")
    @Expose
    String error;

    public String getCode() {
        return code;
    }

    public void setCode(String code) {
        this.code = code;
    }

    public boolean isError() {
        return error != null;
    }

    public String getErrorMessage() {
        return error;
    }
}

File: app/src/main/java/com/chuongvd/app/signal/data/source/remote/service/BaseErrorResponse.kt
package com.chuongvd.app.signal.data.source.remote.service

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

class BaseErrorResponse {

  @SerializedName("code")
  var code: String? = null

  @SerializedName("messageKey")
  @Expose
  var errorMessage: String? = null
    internal set

  val isError: Boolean
    get() = errorMessage != null
}
